recipe-client-addon/lib/NormandyDriver.jsm

```diff
@@ -156,6 +156,47 @@ this.NormandyDriver = function(sandboxManager) {
       sandboxManager.removeHold(`setTimeout-${token}`);
     },
 
+    /*
+     * Return a promise that resolves to an Addon ID if installation is successful.
+     */
+    async installAddon(installUrl) {
+      const installObj = await AddonManager.getInstallForURL(installUrl, null, "application/x-xpinstall");
+      const result = new Promise((resolve, reject) => installObj.addListener({
+        onInstallEnded(addonInstall, addon) {
+          resolve(addon.id);
+        },
+        onInstallFailed(addonInstall) {
+          reject(`AddonInstall error code: [${addonInstall.error}]`);
+        },
+      }));
+      installObj.install();
+      return result;
+    },
+
+    /*
+     * Return a promise that resolves to a success messsage if
+     * addon uninstall is successful.
+     */
+    async uninstallAddon(addonId) {
+      return (id => {
+        return new Promise((resolve, reject) => {
+          AddonManager.getAddonByID(id, addon => {
```

Comment:
This is a nit, but it'd be cleaner to only handle fetching the add-on in the promise, and leaving the rest of the logic outside:

```js
const addon = await new Promise(resolve => AddonManager.getAddonByID(id, resolve));
if (addon === null) {
  throw new Error(`No add-on with ID ${id} found!`);
} else {
  addon.uninstall();
}
```